### DISCUSSION TITLE
API alignment

### DISCUSSION THREAD

**Discussion by Krejtcha** [neo4j-graphql/neo4j-graphql-java#291] [Q&A] [answered]
Hi, I have a question about API alignment with the official [neo4j graphql library](https://neo4j.com/docs/graphql-manual/current/).

In you readme you state that this library should match with the neo4j library as much as possible, but I see some differences and would like to ask if they will be addressed or I can consider them final it this library:

- [filtering and capitalization](https://neo4j.com/docs/graphql-manual/current/filtering/)
  - this library has e.g. `name_contains`, but the neo4j library has `name_CONTAINS`, do you plan to align this?
- [offset-based pagination](https://neo4j.com/docs/graphql-manual/current/pagination/offset-based/)
  - this library has `limit` and `skip`, but the neo4j library has `limit` and `offset`, do you plan to align this?

Thank you for your answer.

**Comment by Andy2003** [comment]
we plan to align this

**Reply by Krejtcha** [reply]
And when can it be expected if I may ask? And will the alignment be done in a form of additional `SchemaConfig` settings or just as a breaking change? Thank you.

**Reply by Andy2003** [reply, marked as answer]
There will be a major release with breaking changes when the alignment is done. I will try to make as much changes configurable so the migration can be as smooth as possible. I cannot give you a concrete time when the alignment is done, since I have only limited resources to work on it.
